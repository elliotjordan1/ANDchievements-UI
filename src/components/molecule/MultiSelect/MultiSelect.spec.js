import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Component from '.';

describe('TechListItem component', () => {
  it('renders correctly with optionList prop', () => {
    const optionList = [1, 2, 3];

    const component = <Component optionList={optionList} />;

    const wrapper = render(component);
    
    expect(wrapper).toMatchSnapshot();
  });
  it('renders correctly with placeholder prop', () => {
    const placeholder = 'Placeholder';

    const component = <Component placeholder={placeholder} />;

    const wrapper = render(component);
    
    expect(wrapper).toMatchSnapshot();
  });
  it('renders correctly with no props', () => {
    const component = <Component />;

    const wrapper = render(component);
    
    expect(wrapper).toMatchSnapshot();
  });
  it('renders correctly with props', () => {
    const placeholder = 'Placeholder';
    const optionList = [1, 2, 3];

    const component = <Component placeholder={placeholder} optionList={optionList} />;

    const wrapper = render(component);
    
    expect(wrapper).toMatchSnapshot();
  });
  it('renders correctly with selected options and adds option successfully', () => {
    const optionList = [
      {
        value: 1,
        label: 'Jess'
      }
    ]

    const onSelect = jest.fn();

    const { component, getByText } = render(<Component visible placeholder="PLACEHOLDER" optionList={optionList} onSelect={onSelect} />);

    expect(component).toMatchSnapshot();

    const input = getByText('Jess');

    expect(onSelect).toHaveBeenCalledTimes(0);

    fireEvent.click(input);

    expect(onSelect).toHaveBeenCalledTimes(1);
  });
  it('renders correctly with selected options and removes option successfully', () => {
    const optionList = [
      {
        andiId: 1,
        andiName: 'Jess'
      }
    ]

    const onRemove = jest.fn();

    const { component, getByText } = render(<Component placeholder="PLACEHOLDER" selectedValues={optionList} onRemove={onRemove} />);

    expect(component).toMatchSnapshot();

    const input = getByText('x');

    expect(onRemove).toHaveBeenCalledTimes(0);

    fireEvent.click(input);

    expect(onRemove).toHaveBeenCalledTimes(1);
  })
});
