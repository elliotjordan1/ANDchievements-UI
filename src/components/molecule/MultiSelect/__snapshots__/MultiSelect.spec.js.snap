// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TechListItem component renders correctly with no props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MultiSelectWrapper-ehyr88-0 PAJQb"
      >
        <div
          class="InputWrapper-b1liev-0 ieRNSH"
        >
          <input
            class="FormInput__Input-sc-1b0ub2b-0 jdgKUs"
            maxlength="40"
            name=""
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MultiSelectWrapper-ehyr88-0 PAJQb"
    >
      <div
        class="InputWrapper-b1liev-0 ieRNSH"
      >
        <input
          class="FormInput__Input-sc-1b0ub2b-0 jdgKUs"
          maxlength="40"
          name=""
          type="text"
          value=""
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`TechListItem component renders correctly with optionList prop 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MultiSelectWrapper-ehyr88-0 PAJQb"
      >
        <div
          class="InputWrapper-b1liev-0 ieRNSH"
        >
          <input
            class="FormInput__Input-sc-1b0ub2b-0 jdgKUs"
            maxlength="40"
            name=""
            type="text"
            value=""
          />
        </div>
        <div
          class="OptionsWrapper__Wrapper-bx0bji-0 hqmxNK"
        >
          <div
            class="MutliSelectOptions__Options-sc-19b20qi-0 cMRzfs"
          />
          <div
            class="MutliSelectOptions__Options-sc-19b20qi-0 cMRzfs"
          />
          <div
            class="MutliSelectOptions__Options-sc-19b20qi-0 cMRzfs"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MultiSelectWrapper-ehyr88-0 PAJQb"
    >
      <div
        class="InputWrapper-b1liev-0 ieRNSH"
      >
        <input
          class="FormInput__Input-sc-1b0ub2b-0 jdgKUs"
          maxlength="40"
          name=""
          type="text"
          value=""
        />
      </div>
      <div
        class="OptionsWrapper__Wrapper-bx0bji-0 hqmxNK"
      >
        <div
          class="MutliSelectOptions__Options-sc-19b20qi-0 cMRzfs"
        />
        <div
          class="MutliSelectOptions__Options-sc-19b20qi-0 cMRzfs"
        />
        <div
          class="MutliSelectOptions__Options-sc-19b20qi-0 cMRzfs"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`TechListItem component renders correctly with placeholder prop 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MultiSelectWrapper-ehyr88-0 PAJQb"
      >
        <div
          class="InputWrapper-b1liev-0 ieRNSH"
        >
          <input
            class="FormInput__Input-sc-1b0ub2b-0 jdgKUs"
            maxlength="40"
            name=""
            placeholder="Placeholder"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MultiSelectWrapper-ehyr88-0 PAJQb"
    >
      <div
        class="InputWrapper-b1liev-0 ieRNSH"
      >
        <input
          class="FormInput__Input-sc-1b0ub2b-0 jdgKUs"
          maxlength="40"
          name=""
          placeholder="Placeholder"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`TechListItem component renders correctly with props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MultiSelectWrapper-ehyr88-0 PAJQb"
      >
        <div
          class="InputWrapper-b1liev-0 ieRNSH"
        >
          <input
            class="FormInput__Input-sc-1b0ub2b-0 jdgKUs"
            maxlength="40"
            name=""
            placeholder="Placeholder"
            type="text"
            value=""
          />
        </div>
        <div
          class="OptionsWrapper__Wrapper-bx0bji-0 hqmxNK"
        >
          <div
            class="MutliSelectOptions__Options-sc-19b20qi-0 cMRzfs"
          />
          <div
            class="MutliSelectOptions__Options-sc-19b20qi-0 cMRzfs"
          />
          <div
            class="MutliSelectOptions__Options-sc-19b20qi-0 cMRzfs"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MultiSelectWrapper-ehyr88-0 PAJQb"
    >
      <div
        class="InputWrapper-b1liev-0 ieRNSH"
      >
        <input
          class="FormInput__Input-sc-1b0ub2b-0 jdgKUs"
          maxlength="40"
          name=""
          placeholder="Placeholder"
          type="text"
          value=""
        />
      </div>
      <div
        class="OptionsWrapper__Wrapper-bx0bji-0 hqmxNK"
      >
        <div
          class="MutliSelectOptions__Options-sc-19b20qi-0 cMRzfs"
        />
        <div
          class="MutliSelectOptions__Options-sc-19b20qi-0 cMRzfs"
        />
        <div
          class="MutliSelectOptions__Options-sc-19b20qi-0 cMRzfs"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`TechListItem component renders correctly with selected options and adds option successfully 1`] = `undefined`;

exports[`TechListItem component renders correctly with selected options and removes option successfully 1`] = `undefined`;
